promql_parser/promql_parser.pyi

```diff
@@ -0,0 +1,269 @@
+"""PromQL Lexer and Parser
+
+The goal of this project is to build a PromQL lexer and parser capable of
+parsing PromQL that conforms with [Prometheus Query](https://prometheus.io/docs/prometheus/latest/querying/basics/).
+
+## Example
+
+The parser entry point is `promql_parser.parse`, which takes a string slice of PromQL
+and returns the parse result as an AST (`promql_parser.Expr`).
+
+```python
+import promql_parser
+
+promql = 'http_requests_total{environment=~"staging|testing|development",method!="GET"} offset 5m'
+expr = promql_parser.parse(promql)
+print(f"Prettify:\n{expr.prettify()}")
+print(f"AST:\n{expr}")
+```
+
+This outputs:
+
+```text
+Prettify:
+http_requests_total{environment=~"staging|testing|development",method!="GET"} offset 5m
+
+AST:
+VectorSelector(VectorSelector { name: Some("http_requests_total"), matchers: Matchers { matchers: [Matcher { op: Re(staging|testing|development), name: "environment", value: "staging|testing|development" }, Matcher { op: NotEqual, name: "method", value: "GET" }] }, offset: Some(Pos(300s)), at: None })
+```
+
+## PromQL compliance
+
+This library declares compatible with [prometheus v2.45.0](https://github.com/prometheus/prometheus/tree/v2.45.0),
+which isreleased at 2023-06-23. Any revision on PromQL after this commit is not guaranteed.
+"""
+
+from datetime import datetime, timedelta
+from enum import Enum
+from typing_extensions import Any, List, Optional, final
```

Comment:
Can we just import them from `typing`? We can bump python version requirement to 3.8 for `typing.final`, 3.7 is EOL anyway.

Reply:
Sure we can, if we indeed bump minimum Python version requirement.